src/librustc_errors/diagnostic.rs

```diff
@@ -232,6 +232,7 @@ impl Diagnostic {
     /// inline it will only show the text message and not the text.
     ///
     /// See `CodeSuggestion` for more information.
+    #[deprecated(note = "Use `span_suggestion_short_with_applicability`")]
```

Comment:
❤️

Reply:
I wonder how this affects other tools that might be using it. They will have to add `#[allow(deprecated)]` for a bit. CC @rust-lang/dev-tools

Reply:
I've [prepared](https://github.com/rust-lang-nursery/rust-clippy/pull/3191) a similar pull request for Clippy (with similar style/formatting mistakes), but it may intersect existing Applicability-assigning pull request there.